webapp/channels/src/packages/mattermost-redux/src/selectors/entities/groups.ts

```diff
@@ -83,6 +83,56 @@ export function getGroupChannels(state: GlobalState, id: string) {
     return getGroupSyncables(state, id).channels;
 }
 
+export const getAllCustomGroups: (state: GlobalState) => Group[] = createSelector(
+    'getAllCustomGroups',
+    getAllGroups,
+    (groups) => {
+        return Object.entries(groups).filter((entry) => (entry[1].allow_reference && entry[1].delete_at === 0 && entry[1].source === GroupSource.Custom)).map((entry) => entry[1]);
+    },
+);
+
+export const getGroupsAssociatedToTeamForReference: (state: GlobalState, teamID: string) => Group[] = createSelector(
```

Comment:
I'm realizing now that these selectors should also be factories because they take parameters, but that's the case for a bunch of the selectors in this file. I feel like we should really be taking a look at all of these as a whole though because most of them seem to only exist to call and be called by other selectors.

I'm fine leaving these new ones as-is though since it feels like fixing that would require reworking a ton of these selectors, and that can be left until after this PR is in